tensorzero-internal/tests/e2e/clickhouse.rs

```diff
@@ -36,680 +40,163 @@ fn get_clean_clickhouse() -> ClickHouseConnectionInfo {
+        }),
+        Box::new(Migration0013 {
+            clickhouse: &clickhouse,
             clean_start: true,
-        })
-        .await
-        .unwrap();
-        clickhouse_migration_manager::run_migration(&Migration0014 { clickhouse })
-            .await
-            .unwrap();
-
-        assert!(!logs_contain("Failed to apply migration"));
-        assert!(!logs_contain("Failed migration success check"));
-        assert!(!logs_contain("Failed to verify migration"));
-
-        assert!(!logs_contain("Applying migration: Migration0000"));
-        assert!(!logs_contain("Migration succeeded: Migration0000"));
-        assert!(!logs_contain("Applying migration: Migration0002"));
-        assert!(!logs_contain("Migration succeeded: Migration0002"));
-        assert!(!logs_contain("Applying migration: Migration0003"));
-        assert!(!logs_contain("Migration succeeded: Migration0003"));
-        assert!(!logs_contain("Applying migration: Migration0004"));
-        assert!(!logs_contain("Migration succeeded: Migration0004"));
-        assert!(!logs_contain("Applying migration: Migration0005"));
-        assert!(!logs_contain("Migration succeeded: Migration0005"));
-        assert!(!logs_contain("Applying migration: Migration0006"));
-        assert!(!logs_contain("Migration succeeded: Migration0006"));
-        assert!(!logs_contain("Applying migration: Migration0008"));
-        assert!(!logs_contain("Migration succeeded: Migration0008"));
-        assert!(!logs_contain("Applying migration: Migration0009"));
-        assert!(!logs_contain("Migration succeeded: Migration0009"));
-        assert!(!logs_contain("Applying migration: Migration0011"));
-        assert!(!logs_contain("Migration succeeded: Migration0011"));
-        assert!(!logs_contain("Applying migration: Migration0013"));
-        assert!(!logs_contain("Migration succeeded: Migration0013"));
-        assert!(logs_contain("Applying migration: Migration0014"));
-        assert!(logs_contain("Migration succeeded: Migration0014"));
-    }
+        }),
+        Box::new(Migration0014 {
+            clickhouse: &clickhouse,
+        }),
+        Box::new(Migration0015 {
+            clickhouse: &clickhouse,
+        }),
+    ];
+
+    // This runs all migrations up to and including the given migration number,
+    // verifying that only the most recent migration is actually applied.
+    let run_migrations_up_to = |migration_num: usize, logs_contain: fn(&str) -> bool| {
+        let migrations = &migrations;
+        async move {
+            // All of the previous migrations should have already been run
+            for (i, migration) in migrations.iter().enumerate().take(migration_num) {
+                let clean_start = clickhouse_migration_manager::run_migration(migration.as_ref())
+                    .await
+                    .unwrap();
+                if i == 0 {
+                    // We know that the first migration was run so clean start should be false
```

Comment:
this comment is misleading, maybe 

`// We know that the first migration was run in a previous test so clean start should be false`